polaris-react/src/components/IndexTable/IndexTable.tsx

```diff
@@ -724,7 +694,7 @@ function IndexTableBase({
   const sharedMarkup = (
     <>
       <EventListener event="resize" handler={handleResize} />
-      <AfterInitialMount>{stickyHeaderMarkup}</AfterInitialMount>
+      {stickyHeaderMarkup}
```

Comment:
This component is just a `useEffect` in disguise.

But we rely on a `useLayoutEffect` to read the DOM and measure the elements rendered by `stickyHeaderMarkup`.

As per [React's docs](https://react.dev/reference/react/useLayoutEffect), `useLayoutEffect` is run _synchronously_ immediately after DOM tree reconciliation (but before paint) whereas `useEffect` is run _a_synchronously at some time in the future.

So the code in `useLayoutEffect` tries to read DOM elements which haven't been rendered yet (and wont be until after the `useEffect` has run), thereby breaking all the things.

Some git investigation reveals that this `useEffect` was inserted to balance out some other code which attempted to set the size of the sticky header elements _on every render_ (so delaying the render was beneficial to allow the regular headings to fully render first).

This PR moves the resizing of sticky header elements to a more appropriate place (inside that `useLayoutEffect`), so there's no need to delay the rendering of the sticky header elements, and hence we don't need `<AfterInitialMount>`'s `useEffect` 🎉